macro/src/lib.rs

```diff
@@ -169,37 +169,34 @@ fn gen_custom_type_method(strct: &ast::CustomType, m: &ast::Method) -> Item {
         );
     }
 
-    let method_invocation = match &m.self_param {
-        Some(_) => {
-            quote! {
-                #this_ident.#method_ident
-            }
-        }
-        None => {
-            quote! {
-                #self_ident::#method_ident
-            }
+    let lifetimes = {
+        let lifetimes = &m.introduced_lifetimes;
+        if lifetimes.is_empty() {
+            quote! {}
+        } else {
+            quote! { <#(#lifetimes),*> }
         }
     };
 
-    match &m.return_type {
-        None => Item::Fn(syn::parse_quote! {
-            #[no_mangle]
-            extern "C" fn #extern_ident(#(#all_params),*) {
-                #method_invocation(#(#all_params_invocation),*);
-            }
-        }),
-        Some(return_typ) => {
-            let return_typ_syn = return_typ.to_syn();
-
-            Item::Fn(syn::parse_quote! {
-                #[no_mangle]
-                extern "C" fn #extern_ident(#(#all_params),*) -> #return_typ_syn {
-                    #method_invocation(#(#all_params_invocation),*)
-                }
-            })
+    let method_invocation = if m.self_param.is_some() {
+        quote! { #this_ident.#method_ident }
+    } else {
+        quote! { #self_ident::#method_ident }
+    };
+
+    let return_tokens = if let Some(return_type) = &m.return_type {
+        let return_type_syn = return_type.to_syn();
+        quote! { -> #return_type_syn }
+    } else {
+        quote! {}
+    };
+
+    Item::Fn(syn::parse_quote! {
+        #[no_mangle]
+        extern "C" fn #extern_ident#lifetimes(#(#all_params),*) #return_tokens {
```

Comment:
nice to see how uncomplicated this part of the change was given the rest of the scaffolding